test/e2e/tests/test_adopted_resource.py

```diff
@@ -0,0 +1,199 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+# 	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+"""Integration tests for the Application Auto Scaling ScalingPolicy API.
+"""
+
+import boto3
+import pytest
+
+from acktest.resources import random_suffix_name
+from acktest.k8s import resource as k8s
+
+from e2e import service_marker, create_adopted_resource
+
+from e2e.replacement_values import REPLACEMENT_VALUES
+from e2e.common.sagemaker_utils import (
+    sagemaker_client,
+    wait_sagemaker_endpoint_status,
+    sagemaker_make_model,
+    sagemaker_make_endpoint_config,
+    sagemaker_make_endpoint,
+)
+from e2e.common.application_autoscaling_utils import (
+    sagemaker_endpoint_register_scalable_target,
+    sagemaker_endpoint_put_scaling_policy,
+    sagemaker_endpoint_deregister_scalable_target,
+    sagemaker_endpoint_delete_scaling_policy,
+)
+
+TARGET_RESOURCE_PLURAL = "scalabletargets"
+POLICY_RESOURCE_PLURAL = "scalingpolicies"
+ENDPOINT_STATUS_INSERVICE = "InService"
+CRD_GROUP = "applicationautoscaling.services.k8s.aws"
+CRD_VERSION = "v1alpha1"
+
+
+@pytest.fixture(scope="module")
+def name_suffix():
+    return random_suffix_name("sagemaker-endpoint", 32)
+
+
+@pytest.fixture(scope="module")
+def applicationautoscaling_client():
+    return boto3.client("application-autoscaling")
+
+
+@pytest.fixture(scope="module")
+def sagemaker_endpoint(name_suffix):
+    model_name = name_suffix + "-model"
+    endpoint_config_name = name_suffix + "-config"
+    endpoint_name = name_suffix
+    resource_id = f"endpoint/{endpoint_name}/variant/variant-1"
+
+    _, _ = sagemaker_make_model(model_name)
+    _, _ = sagemaker_make_endpoint_config(
+        model_name, endpoint_config_name
+    )
+    _, _ = sagemaker_make_endpoint(
+        endpoint_name, endpoint_config_name
+    )
+
+    wait_sagemaker_endpoint_status(endpoint_name, ENDPOINT_STATUS_INSERVICE)
+    assert resource_id is not None
+
+    yield resource_id
+    sagemaker_client().delete_endpoint(EndpointName=endpoint_name)
+    sagemaker_client().delete_endpoint_config(EndpointConfigName=endpoint_config_name)
+    sagemaker_client().delete_model(ModelName=model_name)
+
+
+@pytest.fixture(scope="module")
+def register_scalable_target(sagemaker_endpoint):
+    resource_id = sagemaker_endpoint
+    _ = sagemaker_endpoint_register_scalable_target(
+        resource_id
+    )
+
+    yield resource_id
+    sagemaker_endpoint_deregister_scalable_target(resource_id)
```

Comment:
delete only if exists? because its already being deleted at end of test

Reply:
I will have to run a botot3 describe call to check, seems unnecessary ?

Reply:
won't it throw an error if the resource doesn't exist?

Reply:
It hasn't so far and the resource should never exit at this point - let me check